paf-mvp-core-js/src/model/model-signature.service.ts

```diff
@@ -0,0 +1,63 @@
+import { IDSASigner } from '@core/crypto/digital-signature';
+import { IdsAndPreferences, Seed } from './generated-model';
+import { UnsignedSource } from './model';
+
+export const SIGN_SEP = '\u2063';
+
+/**
+ * Service for signing models
+ */
+export interface IModelSignatureService {
+  /**
+   * Sign a seed. The signature relies on the ids and preferences of the user.
+   */
+  signSeed(seed: UnsignedSource<Seed>, idsAndPreferences: IdsAndPreferences): Promise<Seed>;
+}
+
+/**
+ * Construct string used for generating the signature of entities.
+ */
+export interface ISignedStringBuilder {
+  /**
+   * Generate the string that must be signed for the Source of the a Seed.
+   */
+  buildStringForSigningSeed(seed: UnsignedSource<Seed>, idsAndPreferences: IdsAndPreferences): string;
```

Comment:
Clear advantage of having a dedicated method is to avoid the composite `SeedSignatureContainer` structure 👍